### ISSUE TITLE
perf(cache): Implement LeCaR adaptive LRU+LFU for content cache

### ISSUE THREAD

**Issue by windoliver** [nexi-lab/nexus#876]
## Summary
Implement the LeCaR (Learning Cache Replacement) algorithm that dynamically chooses between LRU and LFU based on workload characteristics.

## Research Background
LeCaR uses online machine learning to adapt cache policy:
- Outperforms ARC by **18x** when cache is small relative to working set
- Uses regret-based learning to balance recency vs frequency
- No manual tuning required

From the research:
> "LeCaR assumes that at every instant, the workload is best handled by a mix of only two fundamental policies: recency-based (LRU) and frequency-based (LFU) evictions."

## How LeCaR Works
```
1. Maintain two ghost caches: LRU-history and LFU-history
2. On cache miss:
   - If evicted item is in LRU-history → increase LRU weight
   - If evicted item is in LFU-history → increase LFU weight
3. Evict using weighted random choice between LRU and LFU
4. Weights adapt via regret minimization
```

## Current Implementation
```python
# storage/content_cache.py
class ContentCache:
    def __init__(self, max_size_bytes: int = 256 * 1024 * 1024):
        self._cache: dict[str, CacheEntry] = {}
        # Uses simple size-based LRU eviction
```

## Proposed Implementation
```python
class LeCaRCache:
    def __init__(self, maxsize: int, learning_rate: float = 0.45):
        self._cache: OrderedDict[K, V] = OrderedDict()
        self._lru_history: deque[K] = deque(maxlen=maxsize)
        self._lfu_history: deque[K] = deque(maxlen=maxsize)
        self._frequency: dict[K, int] = defaultdict(int)
        self._lru_weight = 0.5
        self._learning_rate = learning_rate
    
    def _evict(self) -> None:
        # Choose policy based on learned weights
        if random.random() < self._lru_weight:
            victim = self._evict_lru()
            self._lru_history.append(victim)
        else:
            victim = self._evict_lfu()
            self._lfu_history.append(victim)
    
    def _update_weights(self, key: K) -> None:
        # Regret-based learning
        if key in self._lru_history:
            self._lru_weight = min(1, self._lru_weight + self._learning_rate)
        elif key in self._lfu_history:
            self._lru_weight = max(0, self._lru_weight - self._learning_rate)
```

## Files to Modify
- `src/nexus/storage/content_cache.py` - Replace with LeCaR
- Optionally: Rust implementation for L1 cache

## Expected Impact
- **18x better** than ARC on small caches
- **Self-adapting** to workload changes
- Better for mixed read patterns (some files accessed frequently, others recently)

## References
- [LeCaR Paper (HotStorage 2018)](https://www.usenix.org/system/files/conference/hotstorage18/hotstorage18-paper-vietri.pdf)
- [CACHEUS: Learning Cache Replacement (FAST 2021)](https://www.usenix.org/conference/fast21/presentation/rodriguez)

## Priority
Medium - Requires new algorithm implementation

## Labels
`performance`, `enhancement`
